core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,293 @@
+var _ services.Service = &triggerConnectorHandler{}
+
+func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, lggr logger.Logger) (*triggerConnectorHandler, error) {
+	if connector == nil {
+		return nil, errors.New("missing connector")
+	}
+	handler := &triggerConnectorHandler{
+		Validator:           capabilities.NewValidator[Config, Input, capabilities.TriggerResponse](capabilities.ValidatorArgs{Info: webapiTriggerInfo}),
+		connector:           connector,
+		registeredWorkflows: map[string]webapiTrigger{},
+		lggr:                lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	return handler, nil
+}
+
+// processTrigger iterates over each topic, checking against senders and rateLimits, then starting event processing and responding
+func (h *triggerConnectorHandler) processTrigger(ctx context.Context, gatewayID string, body *api.MessageBody, sender ethCommon.Address, payload webapicapabilities.TriggerRequestPayload) {
+	// Pass on the payload with the expectation that it's in an acceptable format for the executor
+	wrappedPayload, err := values.WrapMap(payload)
+	if err != nil {
+		h.lggr.Errorw("Error wrapping payload")
+		err = h.sendResponse(ctx, gatewayID, body, webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "Error wrapping payload"})
+		if err != nil {
+			h.lggr.Errorw("error sending response", "err", err)
+		}
+		return
+	}
+	topics := payload.Topics
+
+	// empty topics is error for V1
+	if len(topics) == 0 {
+		h.lggr.Errorw("No Matching Workflow Topics")
+		err = h.sendResponse(ctx, gatewayID, body, webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "No Matching Workflow Topics"})
+		if err != nil {
+			h.lggr.Errorw("error sending response", "err", err)
+		}
+		return
+	}
+
+	matchedWorkflows := 0
+	var response webapicapabilities.TriggerResponsePayload
+	for _, trigger := range h.registeredWorkflows {
+		for _, topic := range topics {
+			if trigger.allowedTopics[topic] {
+				matchedWorkflows++
+
+				if !trigger.allowedSenders[sender.String()] {
+					h.lggr.Errorw("Unauthorized Sender", "sender", sender.String(), "messageID", body.MessageId)
+					err = h.sendResponse(ctx, gatewayID, body, webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "Unauthorized Sender"})
+					if err != nil {
+						h.lggr.Errorw("error sending response", "err", err)
+					}
+					return
+				}
+				if !trigger.rateLimiter.Allow(body.Sender) {
+					h.lggr.Errorw("request rate-limited", "sender", sender.String(), "messageID", body.MessageId)
+					err = h.sendResponse(ctx, gatewayID, body, webapicapabilities.TriggerResponsePayload{Status: "ERROR", ErrorMessage: "request rate-limited"})
+					if err != nil {
+						h.lggr.Errorw("error sending response", "err", err)
+					}
+					return
+				}
+
+				TriggerEventID := body.Sender + payload.TriggerEventID + strconv.FormatInt(time.Now().Unix(), 10)
```

Comment:
@DavidOrchard  last week we talked about having this in the Gateway, so it is generated once then told to each node. Let's do that here to have it consistent across nodes.

One thing I'm worried about is leaving it up to the user to ensure changing the TriggerEventID on every request, rather than enforcing it ourselves. https://github.com/smartcontractkit/chainlink/pull/14580#discussion_r1777952976